### PR TITLE
[KnownBits] Check that mul is optimal for low order bits

### PR DESCRIPTION
**Pull request by jayfoad**

### CODE REVIEW

**Review by goldsteinn — APPROVED** [review]

**Comment by goldsteinn** [comment]
LGTM

File: llvm/unittests/Support/KnownBitsTest.cpp [hunk 1]
```diff
@@ -815,4 +815,38 @@ TEST(KnownBitsTest, ConcatBits) {
   }
 }
 
+TEST(KnownBitsTest, MulExhaustive) {
```
Comment (artagnon) [inline llvm/unittests/Support/KnownBitsTest.cpp:818]: Shouldn't this be called `MulLowerBitsExhaustive` or something?

Comment (jayfoad) [reply llvm/unittests/Support/KnownBitsTest.cpp:818]: I don't think the current name is wrong, and I hope to extend it to high bits as well as low bits.

**Review by artagnon — COMMENTED** [review]
LGTM.

**Pull request merged**